spec/models/beta/search/intercept_message_spec.rb

```diff
@@ -1,12 +1,80 @@
 RSpec.describe Beta::Search::InterceptMessage do
   describe '.build' do
-    subject(:result) { described_class.build(search_query_parser_result.original_search_query) }
+    subject(:intercept_message) { described_class.build(search_query) }
 
-    let(:search_query_parser_result) { build(:search_query_parser_result, :multiple_hits) }
+    context 'when the query corresponds to an existing intercept message' do
+      let(:search_query) { 'plasti' }
 
-    it { is_expected.to be_a(described_class) }
-    it { expect(result.id).to eq('f12f2d963bd9edfcbc56138adff3698a') }
-    it { expect(result).to respond_to(:term) }
-    it { expect(result).to respond_to(:message) }
+      it { is_expected.to be_a(described_class) }
+      it { expect(intercept_message.id).to eq('79ad8b533ed4f173c6d08dd1ba89a204') }
+      it { expect(intercept_message).to respond_to(:term) }
+      it { expect(intercept_message).to respond_to(:message) }
+    end
+
+    context 'when the query does not correspond to an intercept message' do
+      let(:search_query) { 'foo' }
+
+      it { is_expected.to be_nil }
+    end
+  end
+
+  describe '#formatted_message' do
+    context 'when there are a mixture of different goods nomenclature to generate links for' do
+      subject(:formatted_message) { build(:intercept_message, :with_mixture_of_goods_nomenclature_to_transform).formatted_message }
+
+      let(:expected_message) do
+        '(chapter 1)[/chapters/01], (heading 0101)[/headings/0101], (subheading 012012)[/subheadings/0120120000-80] and (commodity 0702000007)[/commodities/0702000007].'
+      end
+
+      it { is_expected.to eq(expected_message) }
+    end
+
+    context 'when there are chapters to generate links for' do
+      subject(:formatted_message) { build(:intercept_message, :with_chapters_to_transform).formatted_message }
+
+      let(:expected_message) do
+        'This should point to (ChaPter 99)[/chapters/99] and (chapters 32)[/chapters/32] and (chapters 1)[/chapters/01] but not chapter 19812321 but for (chapter 9)[/chapters/09].'
+      end
+
+      it { is_expected.to eq(expected_message) }
+    end
+
+    context 'when there are headings to generate links for' do
+      subject(:formatted_message) { build(:intercept_message, :with_headings_to_transform).formatted_message }
+
+      let(:expected_message) do
+        'This should point to (hEadIngs 0101)[/headings/0101] and (heading 0102)[/headings/0102] but not heading 2 or heading 012012 but for (heading 0105)[/headings/0105].'
```

Comment:
I guess it was not a mistake, since I can see a similar one
`hEadIngs`

maybe you did it to the insensitiveness for Upcase/downcase ?

Reply:
Exactly